designs/logging_facade/logging_facade.md

```diff
@@ -0,0 +1,147 @@
+# Using SLF4J Logging Facade
+Currently, as of [osc-core / 02b8024](https://github.com/opensecuritycontroller/osc-core/commit/02b80247faa29b069d2b6082f2e866a2f71b0f20),
+OSC uses the **log4j** logging framework directly. The goal is to add flexibility by having all the code reference 
+the **slf4j** facade only. Behind the facade would be **logback**, **log4j** or another logging framework. The development team will be able to make that change with minimal effort.
+
+## Background
+
+The **slf4j** facade provides a common API, including logging levels: **error**, **warn**, **info**, **debug** and (since 1.4.0) **trace**. It detects the presence of one of several logging frameworks and uses it. 
+
+We want to be able to use **slf4j** throughout the project because:
+- It would then be easy to replace the logging framework behind the facade.
+- The implementation log4j-1.2, currently used, is obsolete.
+- A specific requirement is that plugin implementations be able to use our **slf4j** log configuration without any dependency on any of our packages. 
+- Example implementations with **logback** and **log4j** have been added. See **LogComponent** under **osc-ui** and the security manager plugin projects under the two experimental branches, ![log4j_POC](https://github.com/dmitryintel/osc-core/tree/logging) and ![logback_POC](https://github.com/dmitryintel/osc-core/tree/logback).
+
+## Design Changes
+- All the osc java code to only import **org.slf4j** packages, except **osc-server** and **osc-export** bundles. 
+- For **osc-server**, the following changes are required:
+  - The dependencies added to pom.xml: **slf4j-api** plus some implementation dependency. (**slf4j-log4j12**, **logback**, etc.)
+  - The **LogUtil.initLogging()** function is modified to initialize the selected logging framework and register with the context.
+  - **osc-server/bnd.bnd** receives the following header:
+  
+    `Provide-Capability: osgi.service;objectClass=org.slf4j.ILoggerFactory`
+    
+- **osc-export** needs the same pom.xml dependency as **osc-server** and the logging framework must be listed in the \*.bndrun files.
+- Other bundles (**osc-ui** or plugins) have to implement their own utility @Component classes to obtain the *ILoggerFactory* implementation from the OSGi framework and deliver it to the non-OSGi classes.
+
+The **osc-server/Server**, activate method needs to call the following method, for the case of **log4j**:

```

Comment:
I am guessing this will replace the existing call to LogUtil.initLog4j(); right?